### PR TITLE
Embedded templates in config

### PR DESCRIPTION
**Pull request by dadgar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -301,14 +301,21 @@ exec {
...
   // This is the destination path on disk where the source template will render.
   // If the parent directories do not exist, Consul Template will attempt to
   // create them.
   destination = "/path/on/disk/where/template/will/render.txt"
 
+  // This option allows embedding a template in the configuration file rather
+  // then supplying the `source` path to the template file. This is useful for
+  // short templates. This option can is mutually exclusive with the `source`
+  // option.
+  embedded_template = "{{key_or_default \"service/redis/maxconns@east-aws\" \"5\"}}"
```
Comment (sethvargo) [inline README.md]: Kinda 🚲 🏠 but I wonder if there will be confusion that we call it "source" above and "template" here. I think the word "template" is overloaded in "consul template"

Comment (sethvargo) [inline README.md]: I wonder if "contents" might make more sense

File: template/template.go [hunk 2]
```diff
@@ -87,14 +103,16 @@ func (t *Template) Execute(brain *Brain) ([]dep.Dependency, []dep.Dependency, []
 // init reads the template file and initializes required variables.
 func (t *Template) init() error {
 	// Render the template
-	contents, err := ioutil.ReadFile(t.Path)
-	if err != nil {
-		return err
+	if t.Contents == "" {
```
Comment (sethvargo) [inline template/template.go]: Nit-picky, but I'd prefer to reverse this logic:

```go
if t.Path != "" {
  // ...
}
```

File: template/template.go [hunk 3]
```diff
@@ -30,9 +32,18 @@ type Template struct {
 // path. If the template does not exist, an error is returned. During
 // initialization, the template is read and is parsed for dependencies. Any
 // errors that occur are returned.
-func NewTemplate(path, leftDelim, rightDelim string) (*Template, error) {
+func NewTemplate(path, contents, leftDelim, rightDelim string) (*Template, error) {
```
Comment (sethvargo) [inline template/template.go]: I don't like this API anymore (I think it's confusing to have an input that is both a "path" and "contents"). I think NewTemplate should just take the contents as a string and we push the file reading up the stack. What do you think?

File: README.md [hunk 4]
```diff
@@ -301,14 +301,21 @@ exec {
...
   source = "/path/on/disk/to/template.ctmpl"
 
   // This is the destination path on disk where the source template will render.
   // If the parent directories do not exist, Consul Template will attempt to
   // create them.
   destination = "/path/on/disk/where/template/will/render.txt"
 
+  // This option allows embedding a template in the configuration file rather
+  // then supplying the `source` path to the template file. This is useful for
+  // short templates. This option can is mutually exclusive with the `source`
```
Comment (nugend) [inline README.md]: This option <strike>can</strike> is

File: README.md (hunk 1 above)
Comment (dadgar) [inline README.md]: I am fine with `contents` as the map structure name.

File: README.md (hunk 4 above)
Comment (dadgar) [inline README.md]: Nice catch

File: template/template.go (hunk 3 above)
Comment (dadgar) [inline template/template.go]: I would actually keep the path as the `ID` method relies on it to disambiguate two templates with the same content

Comment (dadgar) [inline template/template.go]: You kinda need the Path for de-dupping the same template in the `ID()` method as well

File: README.md (hunk 1 above)
Comment (dadgar) [inline README.md]: Done

File: template/template.go (hunk 2 above)
Comment (dadgar) [inline template/template.go]: Done

**Pull request merged**